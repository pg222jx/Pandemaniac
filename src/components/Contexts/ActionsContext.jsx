import React,{ useState, createContext } from 'react';

export const ActionsContext = createContext()

export const ActionsProvider = props => {

    const [amount, setAmount] = useState(2)

    return (
        <ActionsContext.Provider value={[amount, setAmount]}>
            {props.children}
        </ActionsContext.Provider>
    )
}

