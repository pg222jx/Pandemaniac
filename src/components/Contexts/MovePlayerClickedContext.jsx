import React,{ useState, createContext } from 'react';

export const MovePlayerClickedContext = createContext()

export const MovePlayerClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <MovePlayerClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </MovePlayerClickedContext.Provider>
    );
}

