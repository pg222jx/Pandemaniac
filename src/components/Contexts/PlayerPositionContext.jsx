import React,{ useState, createContext } from 'react';

export const PlayerPositionContext = createContext()

export const PlayerPositionProvider = props => {

    const [position, setPosition] = useState({bottom: undefined, left: undefined})

    return (
        <PlayerPositionContext.Provider value={[position, setPosition]}>
            {props.children}
        </PlayerPositionContext.Provider>
    )
}

