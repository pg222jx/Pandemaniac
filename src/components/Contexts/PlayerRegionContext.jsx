import React,{ useState, createContext } from 'react';

export const PlayerRegionContext = createContext()

export const PlayerRegionProvider = props => {

    const [region, setRegion] = useState('')

    return (
        <PlayerRegionContext.Provider value={[region, setRegion]}>
            {props.children}
        </PlayerRegionContext.Provider>
    )
}

