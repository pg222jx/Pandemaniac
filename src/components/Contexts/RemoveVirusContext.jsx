import React,{ useState, createContext } from 'react';

export const RemoveVirusContext = createContext()

export const RemoveVirusProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <RemoveVirusContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </RemoveVirusContext.Provider>
    );
}

