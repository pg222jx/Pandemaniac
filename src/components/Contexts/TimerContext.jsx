import React,{ useState, createContext } from 'react';

export const TimerContext = createContext()

export const TimerProvider = props => {

    let [time, setTime] = useState(0)

    return (
        <TimerContext.Provider value={[time, setTime]}>
            {props.children}
        </TimerContext.Provider>
    )
}

