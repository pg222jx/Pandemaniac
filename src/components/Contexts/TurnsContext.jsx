import React,{ useState, createContext } from 'react';

export const TurnsContext = createContext()

export const TurnsProvider = props => {

    const [amount, setAmount] = useState(0)

    return (
        <TurnsContext.Provider value={[amount, setAmount]}>
            {props.children}
        </TurnsContext.Provider>
    )
}

