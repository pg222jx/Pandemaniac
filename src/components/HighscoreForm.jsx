/**
 * Sets the highscores to firebase.
 *
 * @author Pernilla GÃ¶th
 * @version 1.0.0
 */


import React, { useState, useContext } from 'react'
import firebase from '../firebase.js'
import { TurnsContext } from './Contexts/TurnsContext'
import { TimerContext } from './Contexts/TimerContext'


const HighscoreForm = () => {
    const [turns, setTurns] = useContext(TurnsContext)// eslint-disable-line no-unused-vars
    const [time, setTime] = useContext(TimerContext)// eslint-disable-line no-unused-vars
    const [nickname, setNickname] = useState('')

  /**
  * Checks for harmful characters.
  * 
  * @returns {boolean} If nickname contains possible harmful characters.
  */
    const isValid = () => {
        const notValid = ['<', '>', '/', '#', '%', '&', '!', '=', '"', ';']
        for (let i = 0; i < notValid.length; i++) {
            if (nickname.includes(notValid[i])) {
                return false
            } 
        }
        return true
    }

    /**
    * Checks Nickname input has correct value.
    */
    const submitHighscore = (e) => {
        e.preventDefault()
        const trimmed = nickname.replace(/\s+/g, '').length
        let nameInfo = document.querySelector('.name_info')
        if (trimmed < 3) {
            nameInfo.textContent = 'Nickname needs to be at least 3 characters.'
        } else if (trimmed > 10) {
            nameInfo.textContent = 'Nickname needs to be egual or less than 10 characters.'
        } else if (isValid() === false) {
            nameInfo.textContent = 'Nickname can\'t contain special characters'
        } else {
            firebase.firestore().collection('highscores').add({
                name: nickname,
                turns: turns,
                time: time
            })
            document.querySelector('.score_info').textContent = 'Now see if you made the highscore list!'
            document.querySelector('.highscore_form').remove()
        }
    }
    
    return (
        <div className="highscore_form_container">
        <form className="highscore_form" onSubmit={submitHighscore}>
            <label>Please enter a nickname for the highscore list:</label><br/><br/>
            <input type="text" className="enterNickname" value={nickname} onChange={e => setNickname(e.target.value)} placeholder="Enter nickname"/>
            <p className="name_info" data-testid="name_info_test"></p>
            <input data-testid="submit_test" className="submit" type="submit" value="Submit"/>
        </form>
        <p className="score_info" data-testid="score_info_test"></p>
        </div>
    )
}

export default HighscoreForm

