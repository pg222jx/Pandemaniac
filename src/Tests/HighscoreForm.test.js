import React from 'react'
import ReactDOM from 'react-dom'
import { render, fireEvent } from '@testing-library/react'
import { shallow } from 'enzyme'
import { TurnsProvider } from '../components/Contexts/TurnsContext'
import { TimerProvider } from '../components/Contexts/TimerContext'
import HighscoreForm from '../components/HighscoreForm'

  const wrapperShallow = shallow(
        <TimerProvider>
            <TurnsProvider>
                <HighscoreForm />
            </TurnsProvider>
        </TimerProvider>
)

  describe('Testing the HighscoreForm component', () => {
    it('should match the snapshot', () => {
        expect(wrapperShallow.html()).toMatchSnapshot()
      })

    it('should render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <TimerProvider>
                <TurnsProvider>
                    <HighscoreForm />
                </TurnsProvider>
            </TimerProvider>, div
        )
    })

    test('so the player are informed when entering a too short nickname.', () => {
        const { getByTestId } = render(
            <TimerProvider>
                <TurnsProvider>
                    <HighscoreForm />
                </TurnsProvider>
            </TimerProvider>
        )

        const submitButton = getByTestId('submit_test')
        const pTag = getByTestId('name_info_test')

        fireEvent.click(submitButton)
        expect(pTag.textContent).toBe('Nickname needs to be at least 3 characters.')
    })

    test('so the input value changes.', () => {
        const { getByPlaceholderText } = render(
            <TimerProvider>
                <TurnsProvider>
                    <HighscoreForm />
                </TurnsProvider>
            </TimerProvider>
        )
        const textForm = getByPlaceholderText('Enter nickname')

        fireEvent.change(textForm, { target: { value: 'qwertyuiopasdfghj' } })
        expect(textForm.value).toBe('qwertyuiopasdfghj')
    })

    test('so the player are informed when entering a too long nickname.', () => {
        const { getByTestId, getByPlaceholderText } = render(
            <TimerProvider>
                <TurnsProvider>
                    <HighscoreForm />
                </TurnsProvider>
            </TimerProvider>
        )

        const submitButton = getByTestId('submit_test')
        const pTag = getByTestId('name_info_test')
        const textForm = getByPlaceholderText('Enter nickname')

        fireEvent.change(textForm, { target: { value: 'qwertyuiopasdfghj' } })
        fireEvent.click(submitButton)
        expect(pTag.textContent).toBe('Nickname needs to be egual or less than 10 characters.')
    })

    test('so the player are informed when entering a correct nickname.', async () => {
        const { getByTestId, getByPlaceholderText } = render(
            <TimerProvider>
                <TurnsProvider>
                    <HighscoreForm />
                </TurnsProvider>
            </TimerProvider>
        )

        const submitButton = getByTestId('submit_test')
        const pTag = getByTestId('score_info_test')
        const textForm = getByPlaceholderText('Enter nickname')

        fireEvent.change(textForm, { target: { value: 'qwertyuiop' } })
        fireEvent.click(submitButton)
        
        await expect(pTag.textContent).toBe('Now see if you made the highscore list!')
    })

    test('so the form is removed if player enters correct nickname.', () => {
        const { getByTestId, getByPlaceholderText } = render(
            <TimerProvider>
                <TurnsProvider>
                    <HighscoreForm />
                </TurnsProvider>
            </TimerProvider>
        )

        let submitButton = getByTestId('submit_test')
        let textForm = getByPlaceholderText('Enter nickname')

        fireEvent.change(textForm, { target: { value: 'qwertyuiop' } })
        fireEvent.click(submitButton)

        expect(submitButton).not.toBeInTheDocument()
        expect(textForm).not.toBeInTheDocument()
    })
  })