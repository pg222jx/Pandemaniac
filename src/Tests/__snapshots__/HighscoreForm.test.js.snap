// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the HighscoreForm component should match the snapshot 1`] = `"<div class=\\"highscore_form_container\\"><form class=\\"highscore_form\\"><label>Please enter a nickname for the highscore list:</label><br/><br/><input type=\\"text\\" class=\\"enterNickname\\" value=\\"\\" placeholder=\\"Enter nickname\\"/><p class=\\"name_info\\" data-testid=\\"name_info_test\\"></p><input type=\\"submit\\" data-testid=\\"submit_test\\" class=\\"submit\\" value=\\"Submit\\"/></form><p class=\\"score_info\\" data-testid=\\"score_info_test\\"></p></div>"`;
